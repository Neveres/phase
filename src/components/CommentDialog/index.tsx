export { default, default as CommentDialog } from './CommentDialog'
