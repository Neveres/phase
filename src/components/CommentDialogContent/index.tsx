export {
  default,
  default as CommentDialogContent,
} from './CommentDialogContent'
