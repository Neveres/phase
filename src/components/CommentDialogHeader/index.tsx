export { default, default as CommentDialogHeader } from './CommentDialogHeader'
