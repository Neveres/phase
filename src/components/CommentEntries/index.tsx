export { default, default as CommentEntries } from './CommentEntries'
