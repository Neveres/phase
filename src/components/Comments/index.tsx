export { default, default as Comments } from './Comments'
