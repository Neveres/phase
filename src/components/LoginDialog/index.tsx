export { default, default as LoginDialog } from './LoginDialog'
