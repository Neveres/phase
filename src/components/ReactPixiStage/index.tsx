export { default, default as ReactPixiStage } from './ReactPixiStage'
