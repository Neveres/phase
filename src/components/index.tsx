export { ReactPixiStage } from './ReactPixiStage'
export { CommentEntries } from './CommentEntries'
export { CommentDialog } from './CommentDialog'
export { AppHeader } from './AppHeader'
export { LoginDialog } from './LoginDialog'
