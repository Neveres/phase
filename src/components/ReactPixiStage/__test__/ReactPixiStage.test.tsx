import React from 'react'
import { create } from 'react-test-renderer'
import { render, waitFor } from '@testing-library/react'
import userEvent from '@testing-library/user-event'
import { ReactPixiStage } from 'src/components'
import rabbit from 'src/assets/rabbit.jpg'
import dog from 'src/assets/dog.jpg'

const resizeWindow = (x: number, y: number) => {
  window.innerWidth = x
  window.innerHeight = y
  window.dispatchEvent(new Event('resize'))
}

describe('ReactPixiStage', () => {
  test('should render well without sprites', () => {
    const testRenderer = create(<ReactPixiStage setCoordinate={jest.fn()} />)
    expect(testRenderer.toJSON()).toMatchSnapshot()
  })

  test('should render well with sprites', () => {
    const sprites = [
      {
        x: 200,
        y: 200,
        anchor: 0.5,
        image: rabbit,
        interactive: true,
        width: 313,
        height: 313,
        draggable: true,
      },
      {
        x: 600,
        y: 200,
        image: dog,
        interactive: true,
        width: 313,
        height: 313,
        anchor: 0.5,
      },
    ]

    const testRenderer = create(
      <ReactPixiStage setCoordinate={jest.fn()} sprites={sprites} />,
    )
    expect(testRenderer.toJSON()).toMatchSnapshot()
  })

  test('click should work well', async () => {
    resizeWindow(500, 500)
    const setCoordinate = jest.fn()
    const sprites = [
      {
        x: 200,
        y: 200,
        anchor: 0.5,
        image: rabbit,
        interactive: true,
        width: 313,
        height: 313,
      },
    ]

    render(<ReactPixiStage setCoordinate={setCoordinate} sprites={sprites} />)
    userEvent.click(document.querySelector('canvas') as any)

    await waitFor(() => {
      expect(setCoordinate).toBeCalled()
    })
  })
})
