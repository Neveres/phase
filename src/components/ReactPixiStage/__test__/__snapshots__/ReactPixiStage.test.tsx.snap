// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ReactPixiStage should render well with sprites 1`] = `
<canvas
  onClick={[Function]}
/>
`;

exports[`ReactPixiStage should render well without sprites 1`] = `
<canvas
  onClick={[Function]}
/>
`;
